cardano-node/src/Cardano/CLI/Genesis.hs

```diff
@@ -0,0 +1,71 @@
+{-# LANGUAGE CPP                        #-}
+{-# LANGUAGE RecordWildCards            #-}
+
+module Cardano.CLI.Genesis
+  ( dumpGenesis
+  )
+where
```

Comment:
I don't think a single function `dumpGenesis` should have its own module. I would put this in `Cardano.CLI.Key` and if possible change the name of the module to encompass this (not strictly necessary for now).

Reply:
Two things to consider:

1. I foresee the `Cardano.CLI.Key` module to expand in future, so it should retain its name, I think.
2. the `dumpGenesis` has its local attendant `writeSecrets` function, so it's not traveling alone
3. I would like to factor the genesis generation into a function as well, so the cutoff point in the `Genesis` subcommand will change -- but again, this PR isn't about factoring -- it's pure simple code motion.

Given this, I don't think it's reasonable to do the module merge.

Reply:
Re: 1. Fore-seeing the expansion of a module is not a good reason to create one. If there are multiple functions that make sense to put in a separate module, then you should create that module. I'm green lighting this PR for now to see where you are going with this. But as a fair warning, later on I may make changes of my own to this structure.

Reply:
First, I don't think that the language of "warnings" (fair or not) is helpful.

I find the language of rationales much less conflict-ful and much more useful.

Next, the export is a single function, indeed.  But so what?  Modules abstract over code.
What matters is that there are two functions in the module, one hidden, and there will be more, fairly soon, I hope.

If we want prior art -- there is a number of modules in `cardano-ledger` that export a single data type and nothing else.  `Cardano.Chain.UTxO.GenesisUTxO`, `Test.Cardano.Chain.MempoolPayload.Gen`, `Test.Cardano.Chain.Config` and `Test.Cardano.Chain.Delegation.Example` export a single function (that isn't called `tests`).

Reply:
Adding a module for a single function adds unnecessary clutter to the directory tree. I'm not sure how else to explain that.

Reply:
It is correct that there is also the FS hierarchy resource that we expend, so that's a cost.

But also important is what we buy with that cost -- and we buy separation of concerns, abstraction -- so we can say "for this bunch of code all the outside world needs to care about is this name".

What are some other benefits?

1. Import statements are topically localised (i.e. import lists from different aspects are not inter-mixed)
1. Locality of code is underscored -- i.e. you can see more clearly the role of ancillary functions, such as `writeSecrets`.
1. We have a ready landing place for future related code.
1. We don't have to come up with painfully contrived or non-descriptive names for modules that combine more than one aspect (i.e. what would you call a module that would cover `CLI.Key` and `CLI.Genesis`?) -- such names are often harder to remember & associate with their content in one's head..
1. Less conflicts.

Yes there are costs.  But it is important to see the balance.